tensornetwork/block_sparse/charge.py

```diff
@@ -641,7 +641,7 @@ def identity_charge() -> np.ndarray:
     def random(cls,
                dimension: int,
                minval: int = 0,
-               maxval: int = n) -> BaseCharge:
+               maxval: int = n - 1) -> BaseCharge:
       if maxval >= n:
         raise ValueError(f"maxval must be less than n={n}, got {maxval}")
       if minval < 0:
```

Comment:
Can we add a test that covers this? Looks like we missed it based on your fix.

Reply:
NP

Reply:
fixed